lib/elixir_bench/github/client.ex

```diff
@@ -1,44 +1,20 @@
 defmodule ElixirBench.Github.Client do
-  alias ElixirBench.Github.Client
+  @moduledoc """
+  The Github client specification.
 
-  defstruct [:base_url, ssl_options: []]
+  This module defines an interface for writing Github clients. The
+  clients provide the api to get the raw configuration based on the
+  benchee.yml file of the target repository. Their implementation can follow
+  different protocol interfaces, like the `ElixirBench.Github.ClientHTTP`
+  module or a client can be defined with testing pourposes, to be used as mock,
+  like in `ElixirBench.Github.ClientInMemory`.
 
-  def get_plain(%Client{} = client, url, params \\ []) do
-    get(client, url(client, url, params), [{"accept", "text/plain"}], &(&1))
-  end
+  A module that implements the Github client behaviour must export:
 
-  def get_json(%Client{} = client, url, params \\ []) do
-    get(client, url(client, url, params), [{"accept", "application/json"}], &Antidote.decode/1)
-  end
+    - a `get_yaml/1` function that takes the url path of the project and returns
+    a map with the configuration found in the benchee.yml that is stored
+    inside the `benchee directory`.
```

Comment:
I feel like this just duplicates information from the function definition/spec further down. If you want to add more information to the function itself, I believe you can add `@doc` aboove the callback definition

Reply:
I removed this.